packages/fs-detectors/src/detect-framework.ts

```diff
@@ -90,3 +150,75 @@ export async function detectFramework({
   );
   return result.find(res => res !== null) ?? null;
 }
+
+/**
+ * Framework with a `detectedVersion` specifying the version
+ * or version range of the relevant package
+ */
+type VersionedFramework = Framework & {
+  detectedVersion?: string;
+};
+
+// Note: Does not currently support a `frameworkList` of monorepo managers
+export async function detectFrameworkRecord({
+  fs,
+  frameworkList,
+}: DetectFrameworkRecordOptions): Promise<VersionedFramework | null> {
+  const result = await Promise.all(
+    frameworkList.map(async frameworkMatch => {
+      const matchResult = await matches(fs, frameworkMatch);
+      if (matchResult) {
+        return {
+          ...frameworkMatch,
+          detectedVersion: matchResult?.detectedVersion,
+        };
+      }
+      return null;
+    })
+  );
+  const frameworkRecord = result.find(res => res !== null) ?? null;
+
+  return frameworkRecord;
+}
+
+export async function detectFrameworkVersion(
+  frameworkRecord: Framework
+): Promise<string | undefined> {
```

Comment:
```suggestion
export function detectFrameworkVersion(
  frameworkRecord: Framework
): string | undefined {
```

Don't think this needs to be async anymore

Reply:
Thanks! [`956af74` (#9009)](https://github.com/vercel/vercel/pull/9009/commits/956af74b3f516e463bcf668d477bce56271e0bd4)